hare/broker.go

```diff
@@ -155,22 +118,31 @@ func (b *Broker) handleMessage(ctx context.Context, msg []byte) error {
 	logger.Debug("broker received hare message")
 
 	msgLayer := hareMsg.Layer
-	if !b.Synced(ctx, msgLayer) {
-		return errNotSynced
-	}
 
-	isEarly := false
-	if err = b.validateTiming(ctx, &hareMsg); err != nil {
-		if !errors.Is(err, errEarlyMsg) {
-			// not early, validation failed
-			logger.With().Debug("broker received a message to a consensus process that is not registered",
-				log.Err(err))
-			return err
+	if !(b.nodeSyncState.IsSynced(ctx) && b.nodeSyncState.IsBeaconSynced(msgLayer.GetEpoch())) {
+		err := fmt.Errorf("broker rejecting message because not synced")
+		logger.With().Debug("rejecting message", log.Err(err))
+		return err
```

Comment:
I don't see a reason to create an error value for things that are used only once, also I have found that error values can deter people from adding context to errors, which can be detrimental.

Reply:
what do you mean used only once? 
this is on the gossip path. it will be called many times before a node eventually become synced.

Reply:
it's mostly because we don't check the error during tests at the moment. if one is testing this method, then specific error would be desired. in any case, not important.